### DISCUSSION TITLE
How do I use HID project with Hoodloader ?

### DISCUSSION THREAD

**Discussion by Julldozer119** [NicoHood/HID#350] [Q&A]
Hi everyone,

First, i'm discovering arduino and I am new to this.

So i decided to make a buttonbox and thought that HID project would be useful. So I got an Arduino MEGA and I installed Hoodloader2.

But here it is... I don't understand how to use HID project now... It seems that the harder part is behind me but I doesnt work.
I cant download a sketch using hidproject directly to the 2560, but if I drop it in the 16U2 i can't use the headers so I dont have any entry. ( I think)

How do I use HID project ? Where Should I drop the sketch or am I missing something ?

Thanks in advance !

**Comment by NicoHood** [comment]
I suggest reading the hoodloader2 wiki carefully

**Reply by Julldozer119** [reply]
Well It does tell me how to upload a sketch in the 16U2 or the 2560 but in the first case it seems I dont have access to the pins of the 2560 and in the second case I can't upload a sketch using HID project in the MCU IO.

**Reply by NicoHood** [reply]
You can. You just need to enter bootloader mode again. It is all described in the wiki

**Reply by Julldozer119** [reply]
I followed the wiki and even in bootloader mode I still have this error :
 #error HID Project can only be used with an USB MCU.
  ^~~~~
exit status 1
Erreur de compilation pour la carte HoodLoader2 Mega 2560

**Reply by NicoHood** [reply]
You can only program the 16u2 with usb functions and the 2560 with all the other stuff. Those 2 MCUs are independant.

**Reply by Julldozer119** [reply]
So I have to write 2 programs ? one for the 16u2 (w/ HIDproject ?) and one for the 2560 ? but how do I make the link between the 2 ?
I saw that HID bridge existed but is it still relevant ? it seems to be 7 years old...

**Reply by NicoHood** [reply]
Yeah, you have to develop your own serial communication protocol
